flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -456,6 +456,49 @@ public class ExecutionConfigOptions {
                             "The max number of async retry attempts to make before task "
                                     + "execution is failed.");
 
+    // ------------------------------------------------------------------------
+    //  Async Table Function
+    // ------------------------------------------------------------------------
+    @Documentation.TableOption(execMode = Documentation.ExecMode.STREAMING)
+    public static final ConfigOption<Integer> TABLE_EXEC_ASYNC_TABLE_MAX_CONCURRENT_OPERATIONS =
+            key("table.exec.async-table.max-concurrent-operations")
```

Comment:
Looks we changed the naming here. @becketqin , should we change in https://github.com/apache/flink/blob/master/flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java#L465 as well to make it consistent? `ml_predict` is not released yet

Reply:
Yes, I think that makes sense. BTW, if we have a FLIP, we should update the FLIP to make it aligned with the code.

Admittedly, changing the config name here causes the inconsistent config name in `AsyncScalarFunction`. We can add the new config key and change the old config keys as deprecated config key.

Another thing is that at this point, our design principle for configs in SQL seems not consistent. Some of the configs are set using hints, while others are in the execution config. It would be good to have a common design principle, so users don't need to guess how to config the physical behaviors.

Reply:
I asked in FLIP-525 voting thread to make the config `table.exec.async-ml-predict.buffer-capacity` to `table.exec.async-ml-predict.max-concurrent-operations`

Reply:
> Reply to the vote thread of the AsyncScalarFunction FLIP to let people know that we want to use the new config name, which is more intuitive. And for backwards compatibility, we are going to make the old key as deprecated key.
If there is no objection, we can update the AsyncScalarFunction FLIP to add the description of this change.

Ok, that all sounds good.  I'll do that.

> Given that these are addition of new things to multiple components. It might make sense to have a new separate FLIP, which aims to align all the SQL physical behavior to support hints.

I agree, that seems good to have a single FLIP if the aim is to make them all consistent.